File: Diagrammatize.Design.Pattern/code/python/Interpreter/Sample/ProgramNode.py

from Node import Node
from CommandListNode import CommandListNode

class ProgramNode(Node):
    def __init__(self):
        self.commandListNode = None

    def parse(self, context):
        context.skipToken("program")
        self.commandListNode = CommandListNode()
        self.commandListNode.parse(context)

    def __str__(self):
        return "[program {}]".format(self.commandListNode)

File: Diagrammatize.Design.Pattern/code/cxx/Interpreter/Sample2/ProgramNode.cpp
#include "ProgramNode.hpp"

ProgramNode::~ProgramNode() {}

void ProgramNode::parse(Context& context) {
    context.skipToken("program");
    commandListNode_.reset(new CommandListNode());
    commandListNode_->parse(context);
}

std::string ProgramNode::toString() {
    return "[program " + commandListNode_->toString() + "]";
}
